src/model/SignedIntent.ts

```diff
@@ -42,23 +43,85 @@ export class SignedIntent {
 
     relay(provider: Provider) {
         if (provider === null || provider === undefined) {
-            throw Error("The provider can not be null or undefined");
             provider = Provider.getGlobal();
-            if (provider === null) throw Error("A valid configuration must be provided or set as global");
+            if (provider === null || provider === undefined) throw Error("A valid configuration must be provided or set as global");
         }
         if (provider.relayer === null || provider.relayer === undefined) {
             throw Error("The provider is invalid, have not relayer");
         }
 
         const relayClient = new RelayClient(provider.relayer);
-        relayClient.post(this);
+        relayClient.post(this).then(response => {
+            if (response.statusCode !== 200) throw Error("The provider is invalid, have not relayer");
+        })
     }
 
-    status(provider: Provider) {
+    async status(provider: Provider): Promise<Status> {
         if (provider === null || provider === undefined) {
-            throw Error("The provider can not be null or undefined");
+            throw Error("The provider can not be null or undefined")
         }
-        // TODO: CODE
+
+        const minConfirmation = 32;
+        const web3 = provider.web3
+        const signatureRelayedEvent = web3.eth.abi.encodeEventSignature('Relayed(bytes32,bytes32[],address,uint256,bytes,bytes32,uint256,bool)')
+        const currentBlockNumber = await web3.eth.getBlockNumber()
+        const data = web3.eth.abi.encodeFunctionCall({
+            name: 'relayedAt',
+            type: 'function',
+            inputs: [{
+                type: 'bytes32',
+                name: '_id'
+            }]
+        }, [this.id])
+        const relayer = web3.eth.abi.encodeFunctionCall({
+            name: 'relayedBy',
+            type: 'function',
+            inputs: [{
+                type: 'bytes32',
+                name: '_id'
+            }]
+        }, [this.id])
+
+        return await web3.eth.call({
```

Comment:
This call is going to fail if the wallet is not yet deployed, is that case contemplated?

Reply:
if wallet is not yet deployed the status is:
Status { code: 'Error' }

Reply:
It should be status code 'pending', to the client is the same a wallet created or not